core/shared/src/main/scala/zio/internal/ConcurrentWeakHashSet.scala

```diff
@@ -198,11 +194,10 @@ private[zio] class ConcurrentWeakHashSet[V](
    *   `true` if the set contains the specified element, `false` otherwise
    */
   override def contains(element: V): Boolean =
-    if (element == null) false
+    if (element eq null) false
     else {
-      val reference     = getReference(element)
-      val storedElement = if (reference ne null) reference.get() else null
-      storedElement != null
+      val reference = getReference(element)
+      (null ne reference) && (null ne reference.get())
```

Comment:
I strongly dislike the `null` first ordering. There isn't a risk of re-assignment.
```suggestion
      (reference ne null) && (reference.get() ne null)
```